### PR TITLE
Document Hot Deploy for Perl and Python images

### PR DESCRIPTION
**Pull request by mnagy**

### CODE REVIEW

File: using_images/s2i_images/perl.adoc [hunk 1]
```diff
@@ -92,3 +92,14 @@ this option while developing or debugging, it is not recommended to turn this
 on in your production environment.
 endif::openshift-origin[]
 |===
+
+ifdef::openshift-origin[]
+== Hot Deploy
+To enable hot deploy in this image, you have to set the `*PERL_APACHE2_RELOAD*` option to "true".
```
Comment (bparees) [inline using_images/s2i_images/perl.adoc]: how do i set that?  is it an environment variable?  doc needs to be clearer.

File: using_images/s2i_images/python.adoc [hunk 2]
```diff
@@ -103,3 +103,19 @@ during the build. Only affects Django projects.
...
 when the produced image is run. Only affects Django projects.
 |===
+
+ifdef::openshift-origin[]
+== Hot Deploy
+If you are using Django, hot deploy will work out of the box.
+
+To enable hot deploy while using Gunicorn, make sure you have a Gunicorn
+configuration file inside your repository with the [`reload`](https://gunicorn-docs.readthedocs.org/en/latest/settings.html#reload) option set to
+`true`. Make sure to specify your config via the `APP_CONFIG` environment
```
Comment (bparees) [inline using_images/s2i_images/python.adoc]: s/config/configuration file/ ?

File: using_images/s2i_images/perl.adoc (hunk 1 above)
Comment (mnagy) [inline using_images/s2i_images/perl.adoc]: Is changing the wording from option to environment variable enough, or should I provide an example or link to oc new-app?

Comment (bparees) [inline using_images/s2i_images/perl.adoc]: changing the wording should be enough, but if we don't already link to the doc about how to set env variables somewhere in this page, we should do so.

Comment (bparees) [inline using_images/s2i_images/perl.adoc]: also it should be documented in this table (which does cover how to setup env variables):
https://docs.openshift.org/latest/using_images/s2i_images/perl.html#configuration

Comment (mnagy) [inline using_images/s2i_images/perl.adoc]: It is mentioned there, but not visible due to an `#ifdef`: https://github.com/openshift/openshift-docs/blame/master/using_images/s2i_images/perl.adoc#L86

**Comment by mnagy** [comment]
Addressed comments, PTAL

**Comment by bparees** [comment]
lgtm, @adellape @CowboysFan ptal/merge.

File: using_images/s2i_images/perl.adoc [hunk 3]
```diff
@@ -92,3 +92,20 @@ this option while developing or debugging, it is not recommended to turn this
...
+
+ifdef::openshift-origin[]
+== Hot Deploy
+To enable hot deploy in this image, you have to set the `*PERL_APACHE2_RELOAD*`
+environment variable to `true`. You can use
+link:../../dev_guide/environment_variables.html#set-environment-variables[oc env]
+to update environment variables of existing objects.
+
+To change your source code in a running Pod, use the
+link:../../cli_reference/basic_cli_operations.adoc#troubleshooting-and-debugging-cli-operations[oc rsh]
```
Comment (adellape) [inline using_images/s2i_images/perl.adoc]: s/adoc/html/

File: using_images/s2i_images/python.adoc [hunk 4]
```diff
@@ -103,3 +103,19 @@ during the build. Only affects Django projects.
 | Set it to a nonempty value to inhibit the execution of `manage.py migrate`
 when the produced image is run. Only affects Django projects.
 |===
+
+ifdef::openshift-origin[]
+== Hot Deploy
+If you are using Django, hot deploy will work out of the box.
+
+To enable hot deploy while using Gunicorn, make sure you have a Gunicorn
+configuration file inside your repository with the [`reload`](https://gunicorn-docs.readthedocs.org/en/latest/settings.html#reload) option set to
```
Comment (adellape) [inline using_images/s2i_images/python.adoc]:
```
https://gunicorn-docs.readthedocs.org/en/latest/settings.html#reload[`reload`]
```

File: using_images/s2i_images/python.adoc [hunk 5]
```diff
@@ -103,3 +103,19 @@ during the build. Only affects Django projects.
...
+`true`. Make sure to specify your configuration file via the `APP_CONFIG`
+environment variable.
+
+To change your source code in a running Pod, use the link:../../cli_reference/basic_cli_operations.adoc#troubleshooting-and-debugging-cli-operations[oc rsh] command to enter the container:
```
Comment (adellape) [inline using_images/s2i_images/python.adoc]: s/adoc/html/

**Comment by mnagy** [comment]
@adellape comments addressed, PTAL

**Comment by adellape** [comment]
Thanks!

**Pull request merged**